### DISCUSSION TITLE
taskiq can't find task. task "broker:add_one" is not found

### DISCUSSION THREAD

**Discussion by sector119** [taskiq-python/taskiq#286] [Q&A]
Hello,

% pwd
/Users/sector119/PycharmProjects/epsilon_tasks
% ls 
Makefile        README          epsilon_tasks   pdm.lock        pyproject.toml

Running worker like "taskiq worker epsilon_tasks.test:broker" I get
[2024-02-06 17:32:56,177][taskiq.worker][INFO   ][MainProcess] Starting 2 worker processes.
[2024-02-06 17:32:56,182][taskiq.process-manager][INFO   ][MainProcess] Started process worker-0 with pid 26371 
[2024-02-06 17:32:56,184][taskiq.process-manager][INFO   ][MainProcess] Started process worker-1 with pid 26372 
task "broker:add_one" is not found. Maybe you forgot to import it?"

What should I change to make taskiq find add_one task?

Thank You

I use example from site

`import asyncio
from taskiq_aio_pika import AioPikaBroker
from taskiq_redis import RedisAsyncResultBackend

broker = AioPikaBroker(
    "amqp://guest:guest@localhost:5672",
).with_result_backend(RedisAsyncResultBackend("redis://localhost"))

@broker.task
async def add_one(value: int) -> int:
    return value + 1

async def main() -> None:
    await broker.startup()
    # Send the task to the broker.
    task = await add_one.kiq(1)
    # Wait for the result.
    result = await task.wait_result(timeout=2)
    print(f"Task execution took: {result.execution_time} seconds.")
    if not result.is_err:
        print(f"Returned value: {result.return_value}")
    else:
        print("Error found while executing task.")
    await broker.shutdown()

if __name__ == "__main__":
    asyncio.run(main())
`

**Comment by s3rius** [comment]
It might be because of different `PYTHONPATH` of your application and the worker. Possible solutions are:
* Run python and taskiq from the same directory so tasks will have same generated names.
* Give name to the task. E.G.

```python
@broker.task(task_name="add_one")
async def add_one(value: int) -> int:
	return value + 1
```

**Reply by sector119** [reply]
Given name to the task do the job. But..

Does tasktiq cache somewhere in rabbitmq tasks or tasks names? I ask because I must restart rabbitmq server to make it work, if I just change name with @broker.task("add_one") and restart worker I get the same error: task "broker:add_one" is not found. Maybe you forgot to import it?

**Reply by s3rius** [reply]
It doesn't cache, but old tasks have generated name and they are in the queue. If you change name and restart the worker, you have to wait untill all old tasks read and discarded.
